src/lib.rs

```diff
@@ -1,8 +1,46 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+use std::net::SocketAddr;
+
+use clap::Parser;
+use tracing_subscriber::filter::LevelFilter;
+
+/// SIFIS-Home wot-rust demo thing
+///
+/// It sets up a servient listening the requested port and address and
+/// advertises itself via mDNS/DNS-SD.
+#[derive(Debug, Parser)]
+pub struct CliCommon {
+    /// Listening port
+    #[arg(short, long, default_value = "3000")]
+    pub listen_port: u16,
+
+    /// Binding address
+    #[arg(short = 'a', long, default_value = "0.0.0.0")]
+    pub bind_addr: std::net::Ipv4Addr,
+
+    /// Verbosity, more output per occurrence
+    #[arg(
+        long,
+        short = 'v',
+        action = clap::ArgAction::Count,
+        global = true,
+    )]
+    verbose: u8,
```

Comment:
Should this be `pub` as well?

Reply:
It is consumed internally, but sure, why not.